src/ui/public/system_api/system_api.js

```diff
@@ -0,0 +1,30 @@
+const SYSTEM_API_HEADER_NAME = 'kbn-system-api';
+
+export default function SystemApisProvider() {
+
+  /**
+   * Adds a custom header designating request as system API
+   * @param headers Object representing set of headers
+   * @return Object representing set of headers, with system API header added in
+   */
+  function addSystemApiHeader(headers) {
+    headers[SYSTEM_API_HEADER_NAME] = true;
```

Comment:
Do you need to mutate the original headers reference, or would a shallow copy be sufficient?